src/tokenized-line.coffee

```diff
@@ -59,7 +74,7 @@ class TokenizedLine
     screenColumn = 0
     currentBufferColumn = 0
     for token in @tokens
-      break if currentBufferColumn > bufferColumn
+      break if currentBufferColumn + token.bufferDelta > bufferColumn
```

Comment:
By the way, I think it would be awesome to add some more tests to document this behavior a bit better. Maybe handling edge cases like the following:

* Left edge of a token
* Middle of a token
* Right of a token

What do you think?